### PR TITLE
Add and run ring workflow 

### PR DESCRIPTION
**Pull request by Sam-Kramer**
## General
**Before this PR**:

**After this PR**:

==COMMIT_MSG==
Runs ring workflow by default in the workload server
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: atlasdb-workload-server-distribution/var/conf/workload-server.timelock.cassandra.yml [hunk 1]
```diff
@@ -43,4 +43,10 @@ install:
       isolationLevel: SERIALIZABLE
     iterationCount: 100
     type: singleRowTwoCells
+  ringConfig:
+    tableConfiguration:
+      tableName: ring_test
```
Comment (Sam-Kramer) [inline atlasdb-workload-server-distribution/var/conf/workload-server.timelock.cassandra.yml:48]: No `-` in Cassandra ❌

File: atlasdb-workload-server-distribution/src/main/java/com/palantir/atlasdb/workload/WorkloadServerLauncher.java [hunk 2]
```diff
@@ -95,12 +99,17 @@ private void runWorkflows(WorkloadServerConfiguration configuration, Environment
                 Refreshable.only(configuration.runtime().atlas()),
                 USER_AGENT,
                 metricsManager);
-        SingleRowTwoCellsWorkflowConfiguration workflowConfig =
+        SingleRowTwoCellsWorkflowConfiguration singleRowTwoCellsConfig =
```
Comment (ergo14) [inline atlasdb-workload-server-distribution/src/main/java/com/palantir/atlasdb/workload/WorkloadServerLauncher.java:102]: Can you rename the PR to reflect that we are adding two workflows and not just the ring workflow? (if I understand correctly!)

File: atlasdb-workload-server-distribution/src/main/java/com/palantir/atlasdb/workload/WorkloadServerLauncher.java [hunk 3]
```diff
@@ -111,11 +120,30 @@ private void runWorkflows(WorkloadServerConfiguration configuration, Environment
...
+        InteractiveTransactionStore transactionStore = transactionStoreFactory.create(
+                Map.of(
+                        workflowConfig.tableConfiguration().tableName(),
+                        workflowConfig.tableConfiguration().isolationLevel()),
+                Set.of());
+        return WorkflowValidator.of(RingWorkflows.create(
+                transactionStore,
+                workflowConfig,
+                MoreExecutors.listeningDecorator(executorService),
+                new AtomicBoolean(false)));
```
Comment (ergo14) [inline atlasdb-workload-server-distribution/src/main/java/com/palantir/atlasdb/workload/WorkloadServerLauncher.java:139]: I missed this in the previous review - but would it make sense to change `RingWorkflows.create` to not take a boolean at all?

It seems like we want skip running to always be false at the start

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/ring/RingWorkflowConfiguration.java [hunk 4]
```diff
@@ -20,14 +20,13 @@
...
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.google.common.util.concurrent.RateLimiter;
-import com.palantir.atlasdb.workload.workflow.ImmutableSingleRowTwoCellsWorkflowConfiguration;
 import com.palantir.atlasdb.workload.workflow.TableConfiguration;
 import com.palantir.atlasdb.workload.workflow.WorkflowConfiguration;
 import org.immutables.value.Value;
 
 @Value.Immutable
-@JsonSerialize(as = ImmutableSingleRowTwoCellsWorkflowConfiguration.class)
-@JsonDeserialize(as = ImmutableSingleRowTwoCellsWorkflowConfiguration.class)
```
Comment (ergo14) [inline atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/ring/RingWorkflowConfiguration.java:30]: oups I missed this too :'(

**Review by ergo14 — CHANGES_REQUESTED** [review]
looks mostly good, just a couple of asks!

File: atlasdb-workload-server-distribution/src/main/java/com/palantir/atlasdb/workload/WorkloadServerLauncher.java (hunk 2 above)
Comment (Sam-Kramer) [reply atlasdb-workload-server-distribution/src/main/java/com/palantir/atlasdb/workload/WorkloadServerLauncher.java:102]: We are actually just adding the ring workflow! Previously we just had the single row two cells workflow

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/ring/RingWorkflowConfiguration.java [hunk 5]
```diff
@@ -38,11 +37,6 @@ default Integer ringSize() {
         return 3;
     }
 
-    @Value.Default
-    default double rateLimit() {
-        return 100;
```
Comment (Sam-Kramer) [inline atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/ring/RingWorkflowConfiguration.java:43]: we probably want this everywhere

File: atlasdb-workload-server-distribution/var/conf/workload-server.timelock.cassandra.yml [hunk 6]
```diff
@@ -45,4 +45,10 @@ install:
...
+      isolationLevel: SERIALIZABLE
```
Comment (Sam-Kramer) [inline atlasdb-workload-server-distribution/var/conf/workload-server.timelock.cassandra.yml:51]: I know we discussed previously this really being a snapshot isolation test, which it is, but for semantics we'll just keep it serializable for now

**Review by ergo14 — APPROVED** [review]
👍

**Pull request merged**